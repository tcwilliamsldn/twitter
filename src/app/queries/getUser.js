import gql from 'graphql-tag';

const getUser = profileName => gql`
    {
        twitter {
            user(identifier: name, identity: "${profileName}") {
                screen_name
                name
                profile_image_url
                tweets_count
                followers_count
                tweets(limit: 50) {
                    id
                    text
                    created_at
                    user {
                        screen_name
                        name
                        profile_image_url
                    }
                }
            }
        }
    }
`;

export default getUser;
