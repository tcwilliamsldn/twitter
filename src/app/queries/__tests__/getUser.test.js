import getUser from '../getTweets';

describe('getUser', function() {
    it('creates gql query with users name', function() {
        const query = getUser('tom');
        expect(query).toMatchSnapshot();
    });
});
