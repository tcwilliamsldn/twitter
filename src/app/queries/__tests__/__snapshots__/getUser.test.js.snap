// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getUser creates gql query with users name 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": undefined,
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "twitter",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "identifier",
                      },
                      "value": Object {
                        "kind": "EnumValue",
                        "value": "name",
                      },
                    },
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "identity",
                      },
                      "value": Object {
                        "block": false,
                        "kind": "StringValue",
                        "value": "tom",
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "user",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "limit",
                            },
                            "value": Object {
                              "kind": "IntValue",
                              "value": "50",
                            },
                          },
                        ],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "tweets",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "created_at",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "retweet_count",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 228,
    "start": 0,
  },
}
`;
